src/Types/HostPort.hs

```diff
@@ -24,12 +25,50 @@ hostPortFromText t = do
     if T.null ptext
       then Right $ HostPort h Nothing
       else do
-        (p,rest) <- decimal $ T.tail ptext
+        (p,rest) <- first (const $ T.unpack $ "Error parsing port in: "<>t) $ decimal $ T.tail ptext
         if T.null rest
           then pure $ HostPort h (Just p)
-          else Left "HostPort encountered invalid port"
+          else Left $ "Could not completely parse port in: " <> T.unpack t
   where
     (h,ptext) = T.break (== ':') t
 
 instance FromJSON HostPort where
   parseJSON = withText "HostPort" $ \v -> either fail pure $ hostPortFromText v
+
+data Scheme = Http | Https
+  deriving (Eq,Ord,Show,Read,Generic)
+
+schemeText :: Scheme -> Text
+schemeText Https = "https"
+schemeText Http = "http"
+
+data SchemeHostPort = SchemeHostPort
+  { _shp_scheme :: Maybe Scheme
```

Comment:
What does a scheme of None mean?

Reply:
It means that the user didn't specify, which gives a better user experience by allowing any of the following to work from the user:

```
localhost
localhost:4443
https://localhost:4443
http://localhost
http://localhost:1848
```

If the user wants to be explicit they can be and if not, then kda-tool can attempt to figure it out in whatever manner is appropriate for the context (which might be different for different commands).

Reply:
Makes sense